src/os_common/os_fhandle.c

```diff
@@ -240,6 +241,20 @@ __wt_open(WT_SESSION_IMPL *session,
 	WT_ERR(__wt_calloc_one(session, &fh));
 	WT_ERR(__wt_strdup(session, name, &fh->name));
 
+	/*
+	 * Mark WiredTiger owned files. To keep code simple, we assume that
+	 * the path separator is one char.
+	 */
+	sep = __wt_path_separator();
+	WT_ASSERT(session, strlen(sep) == 1);
+	fp = strrchr(name, sep[0]);
+	if (fp == NULL)
+		fp = name;
+	else
+		++fp;
+	if (WT_PREFIX_MATCH(fp, "WiredTiger"))
+		F_SET(fh, WT_FH_WIREDTIGER_OWNED);
+
```

Comment:
Why not use `strstr(3)` instead, and skip the magic?

Reply:
We are looking for the final separator, and there is no `strrstr` (reverse string search).

Reply:
I was thinking `strstr` for the separator until you run out of matches, and then a `strcmp` for the `WiredTiger` string.

I just really hate that `WT_ASSERT`.

How about what you have, but after the `strrchr`, do `WT_PREFIX_SKIP` of the separator string, followed by `WT_PREFIX_MATCH` of `WiredTiger`?

Reply:
This discussion is now moot, we no longer look at file names.